api/apps/services/structure_graph_common.py

```
@@ -363,80 +367,203 @@ async def keyword_subgraph(
+        query = text_query.lower()
+        if not name or not query:
+            return False
+        if query in name:
+            return True
+        terms = [term for term in query.split() if term]
+        return bool(terms) and all(term in name for term in terms)
+
+    # Entity names are identifiers in the graph. Prefer lexical matching so
+    # an exact title/entity name wins over a semantically similar row such as
+    # a navigation or summary record.
+    text_query = re.sub(r"[ :|\r\n\t,，。？?/`!！&^%()\[\]{}<>*~'\"\\=]+", " ", str(keywords)).strip()
```

Ruff (0.16.1)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)